logic/hints.py

```diff
@@ -237,37 +238,32 @@ def calculate_possible_barren_regions(worlds: list[World]) -> None:
                     and world.setting("boss_keys").is_any_of("vanilla", "own_dungeon")
                 )
             ):
-                junk_locations.add(location)
-            # Depending on how items were placed, temporarily set certain
+                non_barren_blockers.add(location)
+            # Depending on how items were placed, set certain
             # items as junk items if all of the item's chain locations also
-            # only contain junk
+            # only contain junk or are ultimately not required
             chain_locations = [
                 world.get_location(loc_name)
                 for loc_name in location.current_item.chain_locations
             ]
-            if location.progression and len(chain_locations) > 0:
-                if locations_are_all_junk(chain_locations):
-                    junk_locations.add(location)
-                    logging.getLogger("").debug(f"{location.current_item} is now junk")
-                else:
-                    potentially_junk_locations.add(location)
-
-        # Iterate through all the potentially_junk_locations until
-        # no new junk locations are set
-        new_junk_locations = True
-        while new_junk_locations:
-            new_junk_locations = False
-            for location in potentially_junk_locations:
-                chain_locations = [
-                    world.get_location(loc_name)
-                    for loc_name in location.current_item.chain_locations
-                ]
-                if location.current_item.is_major_item and locations_are_all_junk(
-                    chain_locations
-                ):
-                    new_junk_locations = True
-                    junk_locations.add(location)
-                    logging.getLogger("").debug(f"{location.current_item} is now junk")
+            original_items = []
```

Comment:
I think this calculation of non-barren-blocking locations could result in locations being hinted as `not required` despite them potentially leading to `possibly required` locations. If a bottle is on turning in rattle, but there are other logical bottles available, then the rattle will get marked as `not required` even if the bottle can be marked as `possibly required` which might be a little disingenuous. This is why I previously went with the `while` loop that checked if the item's chain locations were junk instead of removing all the chain location items and then checking game beatability. I'd personally be in favor of reverting the calculation to its previous version.